include/cuco/bit_vector.cuh

```diff
@@ -59,51 +91,100 @@ class bit_vector {
   bit_vector();
   ~bit_vector();
 
-  void add(bool bit);  // adds a new bit at the end
-  void build();        // builds indexes for rank and select.
-
+  /**
+   * @brief adds a new bit at the end
+   *
+   * @param bit Boolean value of new bit to be added
+   */
+  void add(bool bit);
+
+  /**
+   * @brief Builds indexes for rank and select
+   *
+   * Also creates device-side snapshot
+   */
+  void build();
+
+  /**
+   * @brief Modifies a single bit
+   *
+   * @param key position of bit to be modified
+   * @param bit new value of bit
+   */
   void set(Key key, bool bit);
+
+  /**
+   * @brief Sets last bit to specified value
+   *
+   * @param bit new value of last bit
+   */
   void set_last(bool bit);
 
-  static constexpr auto cg_size      = 1;
-  static constexpr auto window_size  = 1;
-  static constexpr auto thread_scope = Scope;
+  static constexpr auto cg_size      = 1;      ///< CG size used to for probing
+  static constexpr auto window_size  = 1;      ///< Window size used to for probing
```

Comment:
So currently there is no point of having a `cg_size` or `window_size` other than one. However, I would say we leave the definition in the class for now, since there might be application in the future.